x/issuance/types/genesis.pb.go

```diff
@@ -0,0 +1,1496 @@
+const _ = proto.GoGoProtoPackageIsVersion3 // please upgrade the proto package
+
+// GenesisState defines the issuance module's genesis state.
+type GenesisState struct {
+	// params defines all the paramaters of the module.
+	Params   Params        `protobuf:"bytes,1,opt,name=params,proto3" json:"params"`
+	Supplies []AssetSupply `protobuf:"bytes,2,rep,name=supplies,proto3" json:"supplies" yaml:"supplies,omitempty"`
+}
+
+func (m *GenesisState) Reset()         { *m = GenesisState{} }
+func (m *GenesisState) String() string { return proto.CompactTextString(m) }
+func (*GenesisState) ProtoMessage()    {}
+func (*GenesisState) Descriptor() ([]byte, []int) {
+	return fileDescriptor_e567e34e5c078b96, []int{0}
+}
+func (m *GenesisState) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *GenesisState) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_GenesisState.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalToSizedBuffer(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *GenesisState) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_GenesisState.Merge(m, src)
+}
+func (m *GenesisState) XXX_Size() int {
+	return m.Size()
+}
+func (m *GenesisState) XXX_DiscardUnknown() {
+	xxx_messageInfo_GenesisState.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_GenesisState proto.InternalMessageInfo
+
+func (m *GenesisState) GetParams() Params {
+	if m != nil {
+		return m.Params
+	}
+	return Params{}
+}
+
+func (m *GenesisState) GetSupplies() []AssetSupply {
+	if m != nil {
+		return m.Supplies
+	}
+	return nil
+}
+
+// Params defines the parameters for the issuance module.
+type Params struct {
+	Assets []*Asset `protobuf:"bytes,1,rep,name=assets,proto3" json:"assets,omitempty" yaml:"assets,omitempty"`
+}
+
+func (m *Params) Reset()      { *m = Params{} }
+func (*Params) ProtoMessage() {}
+func (*Params) Descriptor() ([]byte, []int) {
+	return fileDescriptor_e567e34e5c078b96, []int{1}
+}
+func (m *Params) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *Params) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_Params.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalToSizedBuffer(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *Params) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_Params.Merge(m, src)
+}
+func (m *Params) XXX_Size() int {
+	return m.Size()
+}
+func (m *Params) XXX_DiscardUnknown() {
+	xxx_messageInfo_Params.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_Params proto.InternalMessageInfo
+
+func (m *Params) GetAssets() []*Asset {
+	if m != nil {
+		return m.Assets
+	}
+	return nil
+}
+
+// Asset type for assets in the issuance module
+type Asset struct {
+	Owner            string     `protobuf:"bytes,1,opt,name=owner,proto3" json:"owner,omitempty" yaml:"owner,omitempty"`
```

Comment:
Related a bit to the discussion about pricefeed - do we want to store `Owner`, and `BlockedAddresses` as `sdk.AccAddress` and `[]sdk.AccAddress`? Is there difficulty in generating the types from proto files if we don't use strings? In general, I would prefer to store the address rather than it's string representation, as this feels more brittle.

cc @rhuairahrighairidh @nddeluca thoughts?

Reply:
To be done in follow on PR